doc/man3/EVP_EncryptInit.pod

```diff
@@ -1258,7 +1258,8 @@ EVP_CIPHER_CTX_is_encrypting() returns 1 if the I<ctx> is set up for encryption
 EVP_CIPHER_param_to_asn1() and EVP_CIPHER_asn1_to_param() return greater
 than zero for success and zero or a negative number on failure.
 
-EVP_CIPHER_CTX_rand_key() returns 1 for success.
+EVP_CIPHER_CTX_rand_key() returns 1 for success and zero or a negative number
```

Comment:
Can you prove that this returns negative..
The paths I can see that use this (including OSSL_CIPHER_PARAM_RANDOM_KEY) seem to return 0 or 1

Reply:
Thanks for your review. `EVP_CIPHER_CTX_rand_key` can invoke and return `EVP_CIPHER_CTX_ctrl`, which contains negative return values. 

Why  `EVP_CIPHER_CTX_ctrl` contains negative values:
+ every check of it is in the format like  `EVP_CIPHER_CTX_ctrl <= 0`
+ it contains the code snippet like:
```
   if (ret <= 0)
        goto end;
 
   if (arg < 0)
        return -1;

   if (set_params)
       ret = evp_do_ciph_ctx_setparams(ctx->cipher, ctx->algctx, params);
   else
       ret = evp_do_ciph_ctx_getparams(ctx->cipher, ctx->algctx, params);
   goto end;
```
According to `crypto/evp/evp_local.h`, `evp_do_ciph_ctx_setparams` has the following return values:
```
/*
 * These methods implement different ways to pass a params array to the
 * provider.  They will return one of these values:
 *
 * -2 if the method doesn't come from a provider
 *    (evp_do_param will return this to the called)
 * -1 if the provider doesn't offer the desired function
 *    (evp_do_param will raise an error and return 0)
 * or the return value from the desired function
 *    (evp_do_param will return it to the caller)
 */
```

Reply:
The code does do this....
 ```` 
 end:
    if (ret == EVP_CTRL_RET_UNSUPPORTED) {
        ERR_raise(ERR_LIB_EVP, EVP_R_CTRL_OPERATION_NOT_IMPLEMENTED);
        return 0;
    }
````

So it is picking up one of the cases..
The decision is then if it should return 0 in the case of -2..
I will let another reviewer have an opinion about this..